src/ceph-volume/ceph_volume/util/disk.py

```diff
@@ -29,6 +29,55 @@ def get_partuuid(device):
     return ' '.join(out).strip()
 
 
+def udevadm_info(device):
+    """
+    Capture device information with udevadm, useful in situations where blkid and lsblk aren't
+    producing enough useful output.
+    """
+    command = ['udevadm', 'info', '--query=property', '--name', device]
+    out, err, rc = process.call(command)
+    return _udevadm_parser(out)
+
+
+def _udevadm_parser(output):
```

Comment:
@jan--f this work includes this version of the parser, I am not clear if we decided to keep pursuing PR https://github.com/ceph/ceph/pull/25201 or if that is on hold for now. In any case, I am happy to consume your version if that gets pushed through. 

This helper is needed to determine the partition number so that we can remove partitions accordingly

Reply:
#25201 is not on hold afaik, I just need to deal with the FreeBDS portion.